pkg/abi/typecomponents.go

```diff
@@ -19,13 +19,16 @@ package abi
 import (
 	"context"
 	"fmt"
+	"regexp"
 	"strconv"
 	"strings"
 
 	"github.com/hyperledger/firefly-common/pkg/i18n"
 	"github.com/hyperledger/firefly-signer/internal/signermsgs"
 )
 
+var internalTypeStructExtractor = regexp.MustCompile(`^struct (.*\.)?([^.\[\]]+)(\[\d*\])*$`)
```

Comment:
In what cases will a struct name have a `<something>.` prefix? Is it a package or class name containing the struct definition (but only a max of 1 level deep)?

Reply:
(maybe this is a request to add a comment to this regex)